server/middleware/checkPrisonerInCaseloadMiddleware.ts

```diff
@@ -4,65 +4,97 @@ import NotFoundError from '../utils/notFoundError'
+    // Prisoners with a caseload of TRN (transferring) can be accessed by people with global search or inactive bookings
+    function authenticateTransfer() {
+      const allowedToViewTransfers = globalSearchUser || inactiveBookingsUser
+      return allowInactive && allowedToViewTransfers
+    }
+
+    /*
+     * Prisoners can only be accessed if they are within your caseload
+     * OR
+     * You are a global search user and the route is able to be accessed globally
+     */
+    function authenticateActiveBooking() {
+      if (prisonerBelongsToUsersCaseLoad(prisonerData.prisonId, caseLoads)) {
+        return true
+      }
+
+      return allowGlobal && globalSearchUser
+    }
+
+    // Some routes can only be accessed if the prisoner is within your active caseload
+    function authenticateActiveCaseloadOnly() {
+      return activeCaseLoadId === prisonerData.prisonId
+    }
 
-      return next(
-        addMiddlewareError(
-          req,
-          next,
-          new NotFoundError(`CheckPrisonerInCaseloadMiddleware: Prisoner is inactive [${prisonerData.prisonId}]`),
-        ),
-      )
+    function authenticationError(message: string) {
+      return next(addMiddlewareError(req, next, new NotFoundError(message)))
     }
 
-    if (activeCaseloadOnly && activeCaseLoadId !== prisonerData.prisonId) {
-      return next(
-        addMiddlewareError(
-          req,
-          next,
-          new NotFoundError('CheckPrisonerInCaseloadMiddleware: Prisoner not in active caseload'),
-        ),
-      )
+    if (activeCaseloadOnly) {
+      if (!authenticateActiveCaseloadOnly()) {
+        return authenticationError('CheckPrisonerInCaseloadMiddleware: Prisoner not in active caseload')
+      }
     }
 
-    if (
-      !activeCaseloadOnly &&
-      !prisonerBelongsToUsersCaseLoad(prisonerData.prisonId, caseLoads) &&
-      !(allowGlobal && globalSearchUser)
-    ) {
-      return next(
-        addMiddlewareError(
-          req,
-          next,
-          new NotFoundError('CheckPrisonerInCaseloadMiddleware: Prisoner not in caseloads'),
-        ),
-      )
+    if (restrictedPatient) {
+      if (!authenticateRestictedPatient()) {
+        return authenticationError('CheckPrisonerInCaseloadMiddleware: Prisoner is restricted patient')
+      }
+    } else if (inactiveBooking) {
+      if (prisonerData.prisonId === 'OUT') {
+        if (!authenticateOut()) {
+          return authenticationError(
+            `CheckPrisonerInCaseloadMiddleware: Prisoner is inactive [${prisonerData.prisonId}]`,
+          )
+        }
+      }
+      if (prisonerData.prisonId === 'TRN') {
+        if (!authenticateTransfer()) {
+          return authenticationError(
+            `CheckPrisonerInCaseloadMiddleware: Prisoner is inactive [${prisonerData.prisonId}]`,
+          )
+        }
+      }
+    } else if (!authenticateActiveBooking()) {
+      return authenticationError('CheckPrisonerInCaseloadMiddleware: Prisoner not in caseloads')
     }
```

Comment:
Might be just me but I think it would be easier to read with reduced nesting, something like this

```javascript
if (restrictedPatient && !authenticateRestictedPatient()) {
  return authenticationError('CheckPrisonerInCaseloadMiddleware: Prisoner is restricted patient');
}

if (inactiveBooking) {
  if (prisonerData.prisonId === 'OUT' && !authenticateOut()) {
    return authenticationError(`CheckPrisonerInCaseloadMiddleware: Prisoner is inactive [${prisonerData.prisonId}]`);
  }
  
  if (prisonerData.prisonId === 'TRN' && !authenticateTransfer()) {
    return authenticationError(`CheckPrisonerInCaseloadMiddleware: Prisoner is inactive [${prisonerData.prisonId}]`);
  }
}

if (!inactiveBooking && !authenticateActiveBooking()) {
  return authenticationError('CheckPrisonerInCaseloadMiddleware: Prisoner not in caseloads');
}
```